lib/comm_qmp.cpp

```diff
@@ -301,7 +333,29 @@ void comm_allreduce_min(double* data)
 
 void comm_allreduce_array(double* data, size_t size)
 {
-  QMP_CHECK( QMP_sum_double_array(data, size) );
+  if (!deterministic_reduce) {
```

Comment:
is there a simple way to avoid the code duplication when we fall back to MPI?

Reply:
I've halved the amount of duplication in my next commit.